### PR TITLE
Adds Output Field Types Mergable to Composition

### PR DESCRIPTION
**Pull request by PascalSenn**

### CODE REVIEW

File: spec/Section 4 -- Composition.md [hunk 1]
```diff
@@ -14,6 +14,100 @@ run in sequence to produce the composite execution schema.
...
+
+**Error Code**
+
+OUTPUT_FIELD_TYPES_NOT_MERGEABLE
+
+**Formal Specification**
+
+- Let {fieldsByName} be a map of field lists where the key is the name of a
+  field and the value is a list of fields from mergeable types from different
+  subgraphs with the same name.
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:27]: Source schemas?

File: spec/Section 4 -- Composition.md [hunk 2]
```diff
@@ -14,6 +14,100 @@ run in sequence to produce the composite execution schema.
...
+- for each {fields} in {fieldsByName}
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:28]:
```suggestion
- For each {fields} in {fieldsByName}
```

File: spec/Section 4 -- Composition.md [hunk 3]
```diff
@@ -14,6 +14,100 @@ run in sequence to produce the composite execution schema.
...
+FieldsAreMergeable(fields):
+
+- Given each pair of members {fieldA} and {fieldB} in {fields}:
+  - Let {typeA} be the type of {fieldA}
+  - Let {typeB} be the type of {fieldB}
+  - {SameOutputTypeShape(typeA, typeB)} must be true.
+
+**Explanatory Text**
+
+Fields on mergeable objects or interfaces with that have the same name are
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:40]:
```suggestion
Fields on mergeable objects or interfaces that have the same name are
```

File: spec/Section 4 -- Composition.md [hunk 4]
```diff
@@ -14,6 +14,100 @@ run in sequence to produce the composite execution schema.
...
+type User {
+  birthdate: String
+}
+
+type User {
+  birthdate: String
+}
+```
+
+Fields with different nullability are mergeable, resulting in merged field with
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:56]:
```suggestion
Fields with different nullability are mergeable, resulting in a merged field with
```

**Review by glen-84 — CHANGES_REQUESTED** [review]

**Comment by glen-84** [comment]
PR title:

```diff
-Adds Output Field Types Mergable to Composition
+Adds Output Field Types Mergeable to Composition
```

File: spec/Section 4 -- Composition.md (hunk 1 above)
Comment (PascalSenn) [reply spec/Section 4 -- Composition.md:27]: yep! thanks

**Pull request merged**